### PR TITLE
Fix #2062 - Use default RealmLiveData wrapper.

### LINKED ISSUE #2062: Use default RealmLiveData wrapper
https://academy.realm.io/posts/android-architecture-components-and-realm/

Instead of managing it in ViewModel, use the official method

**Would you like to work on the issue?**
No

### PR DESCRIPTION
**Pull request by mayank8318**
Fixes #2062 

Changes:
Step 1: Implement LiveRealmData for LocationsFragment.
Step 2: Implement LiveRealmData for AboutFragment.
Step 3: Implement LiveRealmData for SpeakerListFragment.
@iamareebjamal Please review and suggest changes. I will repeat that for other view models.

### CODE REVIEW

**Comment by mayank8318** [comment]
App is running in my mobile but Travis says "Compilation failed" ?

**Comment by iamareebjamal** [comment]
```
<===----------> 30% EXECUTING [2m 9s]> :app:compileGoogleplayDebugJavaWithJavac<===----------> 30% EXECUTING [2m 10s]<===----------> 30% EXECUTING [2m 11s]<===----------> 30% EXECUTING [2m 12s]e/travis/build/fossasia/open-event-android/android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java:87: error: cannot find symbol
                .addChangeListener((microlocations, orderedCollectionChangeSet) -> {
                ^
  symbol:   method addChangeListener((microloca[...]; } })
  location: class LiveRealmData<Microlocation>
```

**Comment by iamareebjamal** [comment]
Switch to Google Play variant and compile

**Comment by mayank8318** [comment]
@iamareebjamal Please check :)

File: android/app/src/main/java/org/fossasia/openevent/dbutils/RealmDataRepository.java [hunk 1]
```diff
@@ -563,6 +563,13 @@ public Completable saveEventDates(List<EventDates> eventDates) {
         return realm.where(EventDates.class).findAll();
     }
 
+    /**
+     * Convert RealmResults to LiveReamlData
+     */
+    public static LiveRealmData<Microlocation> asLiveData(RealmResults<Microlocation> data) {
+        return new LiveRealmData<>(data);
+    }
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/dbutils/RealmDataRepository.java]: You can create a generic method instead

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by mayank8318** [comment]
Changes : Use generic method for asLiveData( ).

**Review by iamareebjamal — DISMISSED** [review]

**Comment by mayank8318** [comment]
@iamareebjamal I will repeat it for other view models which are ready.

**Comment by mayank8318** [comment]
Changes : Add wrapper for AboutViewModel.
@iamareebjamal Please review :)

**Comment by iamareebjamal** [comment]
@mayank8318 Can you try this scenario:

- Go to any page with bookmarked items
- Press on an item to get detail view of that item
- Toggle the bookmark of that item
- Return to the previous screen and see if the state changed there as well

**Comment by mayank8318** [comment]
@iamareebjamal State changes in the previous screen but bookmarks are not visible in the about fragment.
dateList is causing the problem I think.
Solved that by making an ASYNC call but now getBookmarkedSessions() is returning empty result.

**Comment by mayank8318** [comment]
Changes : Make async call for date list.
Bookmarks work when I change to another app and open Open-Event again (when it resumes focus).
@iamareebjamal Help needed! Can't find why this is happening.

**Comment by iamareebjamal** [comment]
I'll look into it and get back to you :smile: Thanks for notifying

**Comment by iamareebjamal** [comment]
That's because everything is asynchronous without chaining. The subscription to sessions is done after the dates are loaded, and hence before the bookmarks can be loaded, their LiveData is returned, and then the transform is done on the already subscribed data, meaning no actual event is sent to the view.

It is like - 
return instanceA;

instanceA.subscribe

After few seconds, instanceA is replaced with instanceB containing the actual bookmark events, but it's of no use because view is already subscribed to instanceA. That's why it is extremely important for shared objects to be strictly immutable. I will recommend you to temporarily make the LiveData fields final and then check out the errors to know what I am talking about. Unfortunately, we cannot make the fields final as we need to map them to RealmLiveData lazily but effectively, there should be one and only one assignment statement for a LiveData

**Comment by iamareebjamal** [comment]
We could have synchronized the events easily with RxJava but it is not as easy with LiveData, so we can have a discussion and come up with a solution because there are a lot of interdependent asynchronous querying going on here. The important point here is to share a single object ALWAYS with the view and then responsibly send events in order. 

Event is loaded -> Send to view, attach date listener
Date is loaded -> Send to view, attach session listener
Sessions are loaded -> Send to view

This way it will work

**Comment by mayank8318** [comment]
Yes got it. DateList code is creating problem since I changed it to an Async call. Using the old code for `dateList` makes sense in my view since it doesn't have a change listener anyways and I tried it too ;)
@iamareebjamal Should I push my changes?

**Comment by iamareebjamal** [comment]
Yes @mayank8318

**Comment by mayank8318** [comment]
Changes : Changed dateList loading back to previous version.

**Review by iamareebjamal — DISMISSED** [review]

**Comment by mayank8318** [comment]
Changes : Add LiveRealmData wrapper for SpeakerListFragmentViewModel.
@iamareebjamal Please review :)

**Review by iamareebjamal — APPROVED** [review]

**Comment by iamareebjamal** [comment]
Great, let's convert other ViewModels in separate PRs

**Pull request merged**